test/conformance/CMakeLists.txt

```diff
@@ -4,8 +4,8 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 set(UR_CONFORMANCE_TEST_DIR ${CMAKE_CURRENT_SOURCE_DIR})
-option(UR_TEST_DEVICES_COUNT "Count of devices on which CTS will be run" 1)
-option(UR_TEST_PLATFORMS_COUNT "Count of platforms on which CTS will be run" 1)
+set(UR_TEST_DEVICES_COUNT 1 CACHE STRING "Count of devices on which CTS will be run")
```

Comment:
soo, since this is now used in `conformance` and `adapters` tests, maybe it should be moved one CMake level up...?

and also, description could be updated (not only CTS)

Reply:
I'm also wondering if this is the best mechanism for limiting which devices tests run on, long term I'd expect `ONEAPI_DEVICE_SELECTOR` to be a more flexible and accurate solution.

Reply:
I have changed it. I think we could leave it like that for now.

Reply:
Benie, yeah, I guess that be nice. I can see we have some issues (at least somehow) related to the `ONEAPI_DEVICE_SELECTOR ` topic, so this is a bit larger change. This is good enough for this fix, I guess.

Reply:
Yeah, this fix makes sense for how things stand today.

Reply:
This is pretty much blocking CTS from running in multi-device configurations, and we have at least one now.